f8a_report/cve_helper.py

```diff
@@ -0,0 +1,142 @@
+        ingested = []
+        missed = []
+        # Check whether CVE node is present in graph or not
+        try:
+            for cve_id in cve_ids:
+                gremlin_query = "g.V().has('cve_id', '{}').valueMap();".format(cve_id)
+                payload = {'gremlin': gremlin_query}
+                resp = requests.post(url=self.gremlin_url, json=payload)
+                graph_resp = resp.json()
+                if resp.status_code == 200:
+                    if graph_resp.get('result', {}).get('data', []):
+                        ingested.append(cve_id)
+                    else:
+                        missed.append(cve_id)
+                else:
+                    msg = "Error - CVEGraphValidation failed for CVE: {} with error: {}".format(
+                            cve_id, graph_resp)
+                    raise ValueError(msg)
+
+            return ingested, missed
+
+        except (ValueError, TypeError, requests.exceptions.ConnectionError) as e:
+            raise ValueError('%r' % e)
+
+    def get_fp_cves_count(self, updated_on):
+        """Identify total false positive CVEs that were not mapped correctly."""
+        try:
+            # Create a query to fetch False Positive PRs closed yesterday
+            query = '+type:pr+is:closed+updated:{}'.format(updated_on)
+            cve_json = self.call_github_api(query=query)
+            return cve_json.get('total_count', -1)
+
+        except (ValueError, TypeError) as e:
+            raise ValueError('%r' % e)
+
+    def call_github_api(self, query):
+        """Return the json output from Github APIs."""
+        # Check if we are above github rate limits
+        # If yes, wait till the limit is reset
+        if self.github_rate_limits <= 1 and self.github_rate_limit_reset > 0:
+            wait_time = self.github_rate_limit_reset - int(dt.now().timestamp())
+            logger.info("Github Rate Limits Exceeded. Waiting for {} seconds".format(wait_time))
+            time.sleep(wait_time)
+        try:
+            resp = requests.get(url=self.github_url + query,
+                                headers={"Accept": "application/vnd.github.symmetra-preview+json"})
+            self.github_rate_limits = int(resp.headers.get('X-RateLimit-Remaining', 0))
+            self.github_rate_limit_reset = int(resp.headers.get('X-RateLimit-Reset', -1))
+
+            return resp.json() or None
+
+        except (ValueError, TypeError, requests.exceptions.ConnectionError) as e:
+            logger.error("Error fetching via github API for query: {}".format(query))
+            raise ValueError('%r' % e)
+
+    def get_open_cves_count(self, updated_on):
+        """Get all the open CVE count for the last [2, 7, 30, 365] days."""
+        cve_stats = {"github_stats": {"open_count": {}}}
+        end_date = (dt.strptime(updated_on, "%Y-%m-%d") - timedelta(days=1)).strftime(
+                    "%Y-%m-%d")
+        try:
```

Comment:
@miteshvp moving the try-except inside the loop will make sure that the report for other days will be through even if there is a problem fetching details for one. the except block can be used for logging the error only.